### ISSUE TITLE
CI blocked by Timeout in PyTest in Windows environment

### ISSUE THREAD

**Issue by bobhy** [xonsh/xonsh#3616]
PyTest in the last few PRs had been failing on Python 3.8 in Windows with non-change-related errors after the tests themselves.  But now, the failure has moved right to the start of the run and that failure is cancelling  runs for all the other configurations.  Symptom now is 'timeout', and it is affecting submissions from multiple users... #3613, #3614.

## Timeout error

<details>

```

Successfully installed xonsh-0.9.18
============================= test session starts =============================
platform win32 -- Python 3.8.3, pytest-5.4.3, py-1.8.1, pluggy-0.13.1
rootdir: d:\a\xonsh\xonsh, inifile: setup.cfg
plugins: cov-2.9.0, flake8-1.0.6, timeout-1.3.4, xonsh-0.9.18
timeout: 15.0s
timeout method: thread
timeout func_only: False
collected 4020 items

tests\test_xonsh.xsh ...                                                 [  0%]
tests\test_lib\test_os.xsh s.                                            [  0%]
tests\test_lib\test_subprocess.xsh ssss                                  [  0%]
C:\Miniconda\envs\3.8-xonsh-test\lib\site-packages\_pytest\compat.py:333: PytestDeprecationWarning: The TerminalReporter.writer attribute is deprecated, use TerminalReporter._tw instead at your own risk.
See https://docs.pytest.org/en/latest/deprecations.html#terminalreporter-writer for more information.
  return getattr(object, name, default)

+++++++++++++++++++++++++++++++++++ Timeout ++++++++++++++++++++++++++++++++++++
```
</details>

For reasons that passeth understanding, I can't copy more lines from the test failure log, here's the whole log in a ZIP...
[logs_666.zip](https://github.com/xonsh/xonsh/files/4764302/logs_666.zip)

**Comment by astronouth7303** [comment]
Looking into this a bit, it's hard to conclude exactly what's going  on from the examples I have (this, https://github.com/xonsh/xonsh/runs/763919134?check_suite_focus=true, and https://github.com/xonsh/xonsh/runs/762754052?check_suite_focus=true).

Two of the three timeout inside of `test_subprocess_logical_operators`, which doesn't _seem_ to be doing anything long-lived. (I don't think it's even spawning processes.) The dot log suggests the third example is failing at approximately the same point. However, none of the dot logs show the rest of `test_aliases.py` being run (which I would expect given my understanding of pytest test ordering), so that data might be flawed.

I don't have a windows box, so I'm not sure I could reasonably investigate this further. (And even if I did, I'd have sincere doubts I could replicate this.)

My suggestion is to just extend the timeout and move on.

**Comment by bobhy** [comment]
I have increased timeout to 240, seems to work.  My impression is that all
execution times in github workflows got stretched out in the last few
weeks, fwiw.

**Comment by astronouth7303** [comment]
For the record, @laloch mentioned in gitter that empirically, it looks like pytest-timeout is applying the timeout globally--longer timeouts cause failures in later tests.

pytest-timeout's docs seem to imply the timeout is meant to be per-test (however, it doesn't seem to just come out and say that).

It is possible this is a bug or edge-case in pytest-timeout.

**Comment by laloch** [comment]
> For the record, @laloch mentioned in gitter that empirically, it looks like pytest-timeout is applying the timeout globally--longer timeouts cause failures in later tests.

No, that's not what I meant. I meant that there are test that take (much) longer than `test_subprocess_logical_operators` and that raising the timeout to, say, 60s makes the test suite fail later.

**Comment by laloch** [comment]
Turns out, that 90 seconds is not enough any more :/
https://github.com/xonsh/xonsh/pull/3623/checks?check_run_id=777035068

**Comment by bobhy** [comment]
I say we just remove our timeout entirely.  Let github enforce a timeout
externally if they want to to defend their infrastructure.

**Comment by laloch** [comment]
Yeah, @bobhy may be right. I don't feel like opening another PR with another arbitrary constant adjustment only to find tomorrow, that the constant needs to be changed again. @astronouth7303, what do you think?

**Comment by astronouth7303** [comment]
Deadlock is a mode of test failure, and killing an individual test allows the rest of the suite to continue. But we're definitely hitting problems where some combination of underprovisioned infra plus some weirdness in the timeout code is making our lives annoying.

My other concern is contributor experience: Long CI times are frustrating. If the individual environments are actually running sequentially, waiting for github to kill a job could mean extremely long CI times. However, if each environment is actually running in parallel, this shouldn't be a problem.

**Comment by laloch** [comment]
> My other concern is contributor experience: Long CI times are frustrating.

Always-failing CI is much worse ;)

**Comment by laloch** [comment]
The individual environments do in fact run in parallel, which I think might be the source of the issue. I don't know how Github Actions quotas are set, but if they're per-repo or per-workflow, that might be why our jobs are starving and eventually getting killed.

**Comment by laloch** [comment]
Nope, sadly, I can confirm that running the jobs more sequentially like [this](https://github.com/laloch/xonsh/actions/runs/138646585/workflow) does not help at all.

**Comment by bobhy** [comment]
Agreed, waiting for github's limits to cancel would be frustrating: https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits. And can I manually cancel an action I think is just taking too long?  The only way I know is to push another change...

But the false positive is even more frustrating and more worth avoiding for contributor satisfaction (IMhO).

**Comment by jnoortheen** [comment]
Closing this as we don't get such CI failures from pytest-timeout . Also the test suit is improved
